bigchaindb/backend/util.py

```diff
@@ -0,0 +1,20 @@
+class BackendModuleDispatchRegisterError(Exception):
+    """Raised when there is a problem registering dispatched functions for a
+    backend module"""
+
+
+def make_module_dispatch_registrar(module):
+    def dispatch_wrapper(obj_type):
+        def wrapper(func):
+            func_name = func.__name__
+            try:
+                dispatch_registrar = getattr(module, func_name)
+                return dispatch_registrar.register(obj_type)(func)
+            except AttributeError as ex:
+                raise BackendModuleDispatchRegisterError(
+                    ("'{module}' does not contain a single-dispatchable "
+                     "function named '{func}'. The backend being used has not "
+                     "been implemented correctly!").format(
+                         func=func_name, module=module.__name__)) from ex
+        return wrapper
+    return dispatch_wrapper
```

Comment:
Didn't want to do this because `dispatch_registrar.register` might not work (`dispatch_registrar` might not actually be a single dispatchable function)